src/python/pants/backend/python/goals/repl.py

```diff
@@ -37,12 +37,17 @@ async def create_python_repl_request(repl: PythonRepl, pex_env: PexEnvironment)
         Digest, MergeDigests((requirements_pex.digest, sources.source_files.snapshot.digest))
     )
 
+    args = pex_env.create_argv(
+        repl.in_chroot(requirements_pex.name), python=requirements_pex.python
+    )
+
     chrooted_source_roots = [repl.in_chroot(sr) for sr in sources.source_roots]
-    extra_env = {**pex_env.environment_dict, "PEX_EXTRA_SYS_PATH": ":".join(chrooted_source_roots)}
+    extra_env = {
+        **pex_env.environment_dict(python_configured=requirements_pex.python is not None),
+        "PEX_EXTRA_SYS_PATH": ":".join(chrooted_source_roots),
```

Comment:
Is there a reason we wouldn’t want to add `PEX_EXTRA_SYS_PATH` as one of the keys filled in by `.environment_dict()`?

Reply:
Yes.

1. We can't pre-fill it because it depends on every call-site; it's not uniform. And some call-sites like Black don't ever want it.
2. We can't ergonomically use a kwarg to `pex_env.environment_dict()` because most Python call sites do not call the method. Most use `PexProcess`, which itself calls this method. Only `repl.py` and `run.py` directly call `pex_env.environment_dict()`, as they are preparing `InteractiveProcess`es instead of `PexProcess`es.